File: src/2025-02-27-deepclaude37-rust/go/exercises/practice/food-chain/food_chain.go
package foodchain

import (
	"fmt"
	"strings"
)

// Define the animals and their corresponding second lines
var animals = []struct {
	name       string
	secondLine string
}{
	{name: "fly", secondLine: ""},
	{name: "spider", secondLine: "It wriggled and jiggled and tickled inside her."},
	{name: "bird", secondLine: "How absurd to swallow a bird!"},
	{name: "cat", secondLine: "Imagine that, to swallow a cat!"},
	{name: "dog", secondLine: "What a hog, to swallow a dog!"},
	{name: "goat", secondLine: "Just opened her throat and swallowed a goat!"},
	{name: "cow", secondLine: "I don't know how she swallowed a cow!"},
	{name: "horse", secondLine: "She's dead, of course!"},
}

// Verse returns a single verse of the song for the given verse number
func Verse(v int) string {
	if v < 1 || v > 8 {
		return ""
	}

	i := v - 1
	animal := animals[i]

	// Special case for horse
	if animal.name == "horse" {
		return fmt.Sprintf("I know an old lady who swallowed a %s.\n%s", animal.name, animal.secondLine)
	}

	// Build the verse
	var lines []string
	lines = append(lines, fmt.Sprintf("I know an old lady who swallowed a %s.", animal.name))
	
	if animal.secondLine != "" {
		lines = append(lines, animal.secondLine)
	}

	// Build the chain of animals she swallowed
	for j := i; j > 0; j-- {
		current := animals[j]
		prev := animals[j-1]
		line := fmt.Sprintf("She swallowed the %s to catch the %s", current.name, prev.name)
		
		// Special case for spider
		if prev.name == "spider" {
			line += " that wriggled and jiggled and tickled inside her"
		}
		line += "."
		lines = append(lines, line)
	}

	// Add the final line for all verses except fly
	if animal.name != "fly" {
		lines = append(lines, "I don't know why she swallowed the fly. Perhaps she'll die.")
	} else {
		lines = append(lines, "I don't know why she swallowed the fly. Perhaps she'll die.")
	}

	return strings.Join(lines, "\n")
}

// Verses returns multiple verses of the song
func Verses(start, end int) string {
	if start < 1 || end > 8 || start > end {
		return ""
	}

	var verses []string
	for i := start; i <= end; i++ {
		verses = append(verses, Verse(i))
	}

	return strings.Join(verses, "\n\n")
}

// Song returns the complete song
func Song() string {
	return Verses(1, 8)
}

File: src/2025-02-27-deepclaude37-rust/java/exercises/practice/food-chain/src/main/java/FoodChain.java
class FoodChain {
    private static final String[][] ANIMALS = {
        {"fly", "", "I don't know why she swallowed the fly. Perhaps she'll die."},
        {"spider", "It wriggled and jiggled and tickled inside her.", "She swallowed the spider to catch the fly."},
        {"bird", "How absurd to swallow a bird!", "She swallowed the bird to catch the spider that wriggled and jiggled and tickled inside her."},
        {"cat", "Imagine that, to swallow a cat!", "She swallowed the cat to catch the bird."},
        {"dog", "What a hog, to swallow a dog!", "She swallowed the dog to catch the cat."},
        {"goat", "Just opened her throat and swallowed a goat!", "She swallowed the goat to catch the dog."},
        {"cow", "I don't know how she swallowed a cow!", "She swallowed the cow to catch the goat."},
        {"horse", "She's dead, of course!", ""}
    };

    String verse(int verse) {
        if (verse < 1 || verse > 8) {
            throw new IllegalArgumentException("Verse must be between 1 and 8");
        }

        StringBuilder sb = new StringBuilder();
        int index = verse - 1;
        
        // First line of each verse
        sb.append("I know an old lady who swallowed a ").append(ANIMALS[index][0]).append(".\n");
        
        // Second line (description of the animal or action)
        if (!ANIMALS[index][1].isEmpty()) {
            sb.append(ANIMALS[index][1]).append("\n");
        }
        
        // For the horse verse, we're done after the description
        if (verse == 8) {
            return sb.toString();
        }
        
        // Add the chain of animals in reverse order
        for (int i = index; i >= 0; i--) {
            if (!ANIMALS[i][2].isEmpty()) {
                sb.append(ANIMALS[i][2]);
                if (i > 0) {
                    sb.append("\n");
                }
            }
        }
        
        return sb.toString();
    }

    String verses(int startVerse, int endVerse) {
        StringBuilder sb = new StringBuilder();
        for (int i = startVerse; i <= endVerse; i++) {
            sb.append(verse(i));
            if (i < endVerse) {
                sb.append("\n\n");
            }
        }
        return sb.toString();
    }
}
